lib/logitech_receiver/hidpp20.py

```diff
@@ -357,53 +359,276 @@ def __len__(self):
+
+    def set_diverted(self, value: bool):
+        """If set, the control is diverted temporarily and reports presses as HID++ events
+        until a HID++ configuration reset occurs."""
+        self._setCidReporting(divert=value)
+
+    def set_persistently_diverted(self, value: bool):
+        """If set, the control is diverted permanently and reports presses as HID++ events."""
+        self._setCidReporting(persist=value)
+
+    def set_rawXY_reporting(self, value: bool):
+        """If set, the mouse reports all its raw XY events while this control is pressed
+        as HID++ events. Gets cleared on a HID++ configuration reset."""
+        self._setCidReporting(rawXY=value)
+
+    def remap(self, to: _NamedInt):
+        """Remaps this control to another action."""
+        self._setCidReporting(remap=int(to))
+
+    def _getCidReporting(self):
+        try:
+            mapped_data = feature_request(
+                self._device,
+                FEATURE.REPROG_CONTROLS_V4,
+                0x20,
+                *tuple(_pack('!H', self._cid)),
+            )
+            if mapped_data:
+                cid, mapping_flags, mapped_to = _unpack('!HBH', mapped_data[:5])
+                if cid != self._cid and _log.isEnabledFor(_WARNING):
+                    _log.warn(
+                        f'REPROG_CONTROLS_V4 endpoint getCidReporting on device {self._device} replied ' +
+                        f'with a different control ID ({cid}) than requested ({self._cid}).'
+                    )
+                self._mapping_flags = mapping_flags
+                self._mapped_to = mapped_to if mapped_to != 0 else self._cid
+            else:
+                raise FeatureCallError('No reply from device.')
+        except Exception:
+            if _log.isEnabledFor(_ERROR):
+                _log.error(f'Exception in _getCidReporting on device {self._device}: ', exc_info=1)
+                # Clear flags and set mapping target to self as fallback
+            self._mapping_flags = 0
+            self._mapped_to = self._cid
+
+    def _setCidReporting(self, divert=None, persist=None, rawXY=None, remap=0):
+        """Sends a `setCidReporting` request with the given parameters to the control. Raises
+        an exception if the parameters are invalid.
+        """
+        if rawXY:
+            # We need diversion to report raw XY, so divert temporarily
+            # (since XY reporting is also temporary)
+            divert = True
+
+        if divert is not None and special_keys.KEY_FLAG.divertable not in self.flags:
+            raise FeatureNotSupported(f'Tried to divert non-divertable control {self.key} ' + f'on device {self._device}.')
+        if persist is not None and special_keys.KEY_FLAG.persistently_divertable not in self.flags:
+            raise FeatureNotSupported(
+                'Tried to persistently divert non-persistently-divertable control ' + f'{self.key} on device {self._device}.'
```

Comment:
f-strings push Solaar to Python 3.6.   I think this is OK, but @FFY00 should also approve.
As well, docs/installation.md needs be changed as it currently says 3.2.

Reply:
I approve.

Reply:
But if we do it, we should start using it moving forward. This patch also uses the `'' % ()` format.

If anyone has time, you can replace the current `%` and `.format()` usages with f-strings.

Reply:
I'll see if there is a good place to put a short paragraph (or document) on coding style in the Solaar documentation.

Reply:
@FFY00 I just noticed that setup.py wants 3.5 but the install document says 3.2.   I'll update both to 3.6.